File: MVC y DAO/src/main/java/co/edu/unisalle/crud/view/EstudianteView.java
package co.edu.unisalle.crud.view;

import co.edu.unisalle.crud.model.Estudiante;

import java.util.List;
import java.util.Optional;
import java.util.Scanner;

/**
 * Vista del patrón MVC.
 *
 * SRP: única responsabilidad — interactuar por consola con el usuario.
 * No conoce ni persistencia ni reglas de negocio.
 */
public class EstudianteView {

    private final Scanner scanner = new Scanner(System.in);

    public int mostrarMenu() {
        System.out.println();
        System.out.println("===== CRUD ESTUDIANTES (MVC + DAO) =====");
        System.out.println("1. Crear estudiante");
        System.out.println("2. Listar estudiantes");
        System.out.println("3. Buscar estudiante por id");
        System.out.println("4. Actualizar estudiante");
        System.out.println("5. Eliminar estudiante");
        System.out.println("0. Salir");
        System.out.print("Opción: ");
        return leerEntero();
    }

    public Estudiante leerEstudiante() {
        System.out.print("Id: ");
        int id = leerEntero();
        System.out.print("Nombre: ");
        String nombre = scanner.nextLine();
        System.out.print("Apellido: ");
        String apellido = scanner.nextLine();
        System.out.print("Correo: ");
        String correo = scanner.nextLine();
        System.out.print("Programa: ");
        String programa = scanner.nextLine();
        return new Estudiante(id, nombre, apellido, correo, programa);
    }

    public int leerId() {
        System.out.print("Id: ");
        return leerEntero();
    }

    public void mostrarEstudiante(Estudiante estudiante) {
        System.out.println(estudiante);
    }

    public void mostrarEstudiante(Optional<Estudiante> estudiante) {
        estudiante.ifPresentOrElse(
                this::mostrarEstudiante,
                () -> mostrarMensaje("No se encontró el estudiante."));
    }

    public void mostrarEstudiantes(List<Estudiante> estudiantes) {
        if (estudiantes.isEmpty()) {
            mostrarMensaje("No hay estudiantes registrados.");
            return;
        }
        estudiantes.forEach(this::mostrarEstudiante);
    }

    public void mostrarMensaje(String mensaje) {
        System.out.println(mensaje);
    }

    private int leerEntero() {
        while (true) {
            String linea = scanner.nextLine();
            try {
                return Integer.parseInt(linea.trim());
            } catch (NumberFormatException ex) {
                System.out.print("Valor inválido, ingrese un número entero: ");
            }
        }
    }
}

File: MVC y DAO/CrudMvcDao/View/EstudianteView.cs
using CrudMvcDao.Model;

namespace CrudMvcDao.View;

/// <summary>
/// Vista del patrón MVC.
///
/// SRP: única responsabilidad — interactuar por consola con el usuario.
/// No conoce ni persistencia ni reglas de negocio.
/// </summary>
public class EstudianteView
{
    public int MostrarMenu()
    {
        Console.WriteLine();
        Console.WriteLine("===== CRUD ESTUDIANTES (MVC + DAO) =====");
        Console.WriteLine("1. Crear estudiante");
        Console.WriteLine("2. Listar estudiantes");
        Console.WriteLine("3. Buscar estudiante por id");
        Console.WriteLine("4. Actualizar estudiante");
        Console.WriteLine("5. Eliminar estudiante");
        Console.WriteLine("0. Salir");
        Console.Write("Opción: ");
        return LeerEntero();
    }

    public Estudiante LeerEstudiante()
    {
        Console.Write("Id: ");
        var id = LeerEntero();
        Console.Write("Nombre: ");
        var nombre = Console.ReadLine() ?? string.Empty;
        Console.Write("Apellido: ");
        var apellido = Console.ReadLine() ?? string.Empty;
        Console.Write("Correo: ");
        var correo = Console.ReadLine() ?? string.Empty;
        Console.Write("Programa: ");
        var programa = Console.ReadLine() ?? string.Empty;
        return new Estudiante(id, nombre, apellido, correo, programa);
    }

    public int LeerId()
    {
        Console.Write("Id: ");
        return LeerEntero();
    }

    public void MostrarEstudiante(Estudiante? estudiante)
    {
        if (estudiante is null)
            MostrarMensaje("No se encontró el estudiante.");
        else
            Console.WriteLine(estudiante);
    }

    public void MostrarEstudiantes(IReadOnlyList<Estudiante> estudiantes)
    {
        if (estudiantes.Count == 0)
        {
            MostrarMensaje("No hay estudiantes registrados.");
            return;
        }
        foreach (var e in estudiantes)
            Console.WriteLine(e);
    }

    public void MostrarMensaje(string mensaje) => Console.WriteLine(mensaje);

    private static int LeerEntero()
    {
        while (true)
        {
            var linea = Console.ReadLine();
            if (int.TryParse(linea, out var n))
                return n;
            Console.Write("Valor inválido, ingrese un número entero: ");
        }
    }
}
